x/blockdb/config.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package blockdb
+
+import "errors"
+
+// DefaultMaxDataFileSize is the default maximum size of the data block file in bytes (500GB).
+const DefaultMaxDataFileSize = 500 * 1024 * 1024 * 1024
+
+// DefaultMaxDataFiles is the default maximum number of data files descriptors cached.
+const DefaultMaxDataFiles = 10
+
+// DatabaseConfig contains configuration parameters for BlockDB.
+type DatabaseConfig struct {
+	// IndexDir is the directory where the index file is stored.
+	IndexDir string
+
+	// DataDir is the directory where the data files are stored.
+	DataDir string
+
+	// MinimumHeight is the lowest block height tracked by the database.
+	MinimumHeight uint64
+
+	// MaxDataFileSize sets the maximum size of the data block file in bytes.
+	MaxDataFileSize uint64
+
+	// MaxDataFiles is the maximum number of data files descriptors cached.
+	MaxDataFiles int
+
+	// CheckpointInterval defines how frequently (in blocks) the index file header is updated (default: 1024).
+	CheckpointInterval uint64
+
+	// SyncToDisk determines if fsync is called after each write for durability.
+	SyncToDisk bool
+
+	// Truncate determines if existing data should be truncated when opening the database.
```

Comment:
I don't think this option is something we would ever use in a production setting, so probably we should remove it.